api/src/test/java/org/pmiops/workbench/reporting/ReportingSnapshotServiceTest.java

```diff
@@ -0,0 +1,144 @@
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class ReportingSnapshotServiceTest {
+  private static final String FAMILY_NAME = "Bobberson";
+  private static final String CONTACT_EMAIL = "bob@example.com";
+  private static final String PRIMARY_EMAIL = "bob@researchallofus.org";
+  private static final String ORGANIZATION = "Test";
+  private static final String CURRENT_POSITION = "Tester";
+  private static final String RESEARCH_PURPOSE = "To test things";
+  private static final long NOW_EPOCH_MILLI = 1594404482000L;
+
+  @MockBean private Random mockRandom;
+  @MockBean private UserService mockUserService;
+  @MockBean private WorkspaceService mockWorkspaceService;
+
+  @Autowired private ReportingSnapshotService reportingSnapshotService;
+
+  private static final TestMockFactory TEST_MOCK_FACTORY = new TestMockFactory();
+
+  @TestConfiguration
+  @Import({
+    CommonMappers.class,
+    ReportingMapperImpl.class,
+    ReportingSnapshotServiceImpl.class,
+    ReportingUploadServiceImpl.class
+  })
+  @MockBean({BigQueryService.class})
+  public static class config {
+    @Bean
+    public Clock getClock() {
+      return new FakeClock(Instant.ofEpochMilli(NOW_EPOCH_MILLI));
+    }
+  }
+
+  @Before
+  public void setup() {
+    // Return "random" numbers 100, 101, 102...
+    doAnswer(
+            new Answer<Long>() {
+              private long lastValue = 100;
+
+              public Long answer(InvocationOnMock invocation) {
+                return lastValue++;
+              }
+            })
+        .when(mockRandom)
+        .nextLong();
+  }
+
+  public void mockWorkspaces() {
+    final DbWorkspace dbWorkspace1 = stubDbWorkspace("aou-rw-123456", "A Tale of Two Cities", 101L);
+    final DbWorkspace dbWorkspace2 = stubDbWorkspace("aou-rw-789", "Moby Dick", 202L);
+
+    doReturn(ImmutableList.of(dbWorkspace1, dbWorkspace2))
+        .when(mockWorkspaceService)
+        .getAllActiveWorkspaces();
+  }
+
+  @Test
+  public void testGetSnapshot_noEntries() {
+    final ReportingSnapshot snapshot = reportingSnapshotService.takeSnapshot();
+    assertThat(snapshot.getCaptureTimestamp()).isEqualTo(NOW_EPOCH_MILLI);
+    assertThat(snapshot.getResearchers()).isEmpty();
+    assertThat(snapshot.getWorkspaces()).isEmpty();
+  }
+
+  @Test
+  public void testGetSnapshot_someEntries() {
+    mockUsers();
+    mockWorkspaces();
+
+    final ReportingSnapshot snapshot = reportingSnapshotService.takeSnapshot();
+    assertThat((double) snapshot.getCaptureTimestamp()).isWithin(100.0).of(NOW_EPOCH_MILLI);
+    assertThat(snapshot.getResearchers()).hasSize(2);
+    assertThat(snapshot.getResearchers().get(0).getResearcherId()).isEqualTo(101);
+    assertThat(snapshot.getResearchers().get(0).getIsDisabled()).isFalse();
```

Comment:
nit: add other properties for completeness
```suggestion
    assertThat(snapshot.getResearchers().get(0).getResearcherId()).isEqualTo(101);
    assertThat(snapshot.getResearchers().get(0).getIsDisabled()).isFalse();
    assertThat(snapshot.getResearchers().get(0).getUsername()).isEqualTo("bob@researchallofus.org");
    assertThat(snapshot.getResearchers().get(0).getFirstName()).isEqualTo("Marge");
```